### PR TITLE
Issue #SB-13158: Code Push Capability on Mobile

### PR DESCRIPTION
**Pull request by sharathkashyap**

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -126,8 +126,23 @@ export class MyApp implements OnInit, AfterViewInit {
       this.handleBackButton();
       this.appRatingService.checkInitialDate();
       this.getUtmParameter();
+      this.checkForCodeUpdates();
     });
   }
+  checkForCodeUpdates() {
+    SunbirdSdk.instance.sharedPreferences.getString("deploymentKey").toPromise().then(deploymentKey => {
```
Comment (swayangjit) [inline src/app/app.component.ts]: @sharathkashyap  We can use preferences ( defined in the constructor) instead of  SunbirdSdk.instance.sharedPreferences.

Comment (sharathkashyap) [reply src/app/app.component.ts]: I have made appropriate changes

File: src/app/app.component.ts [hunk 2]
```diff
@@ -126,8 +126,57 @@ export class MyApp implements OnInit, AfterViewInit {
       this.handleBackButton();
       this.appRatingService.checkInitialDate();
       this.getUtmParameter();
+      this.checkForCodeUpdates();
+      this.platform.resume.subscribe(() => this.checkForCodeUpdates());
     });
   }
+  checkForCodeUpdates() {
+    this.preferences.putString(PreferenceKey.DEPLOYMENT_KEY,"agojO-OZt4dZlt_pu9r9j2Ipy_jY90dbb065-3633-45a5-9c55-c0405eafaebb").toPromise().then();
```
Comment (swayangjit) [inline src/app/app.component.ts]: We can put the deployment key in the properties file and access it from there.

File: src/app/app.component.ts [hunk 3]
```diff
@@ -126,8 +126,57 @@ export class MyApp implements OnInit, AfterViewInit {
       this.handleBackButton();
       this.appRatingService.checkInitialDate();
       this.getUtmParameter();
+      this.checkForCodeUpdates();
+      this.platform.resume.subscribe(() => this.checkForCodeUpdates());
```
Comment (swayangjit) [inline src/app/app.component.ts]: We are already subscribing for platform resume in line no 276. we can move the code to that block.

Comment (swayangjit) [reply src/app/app.component.ts]: is it required to check for updates always when the app resumes?

File: src/app/app.component.ts (hunk 2 above)
Comment (sharathkashyap) [reply src/app/app.component.ts]: I think this is a temporary implementation which will be removed eventually in next iteration. This is just to inform notification Handling User Story owner to use this statement as part of notification handling

File: src/app/app.component.ts (hunk 3 above)
Comment (sharathkashyap) [reply src/app/app.component.ts]: Yes Swayanjgit we may need to check for every update

**Pull request merged**